pytorch_lightning/utilities/fetching.py

```diff
@@ -174,35 +208,183 @@ class DataFetcher(AbstractDataFetcher):
 
     """
     This class is used to control batch fetching flow.
+
+    By default, the `fetching_function` will `prefetch` a batch in advance to detect the end of the iteration.
     """
 
+    @contextlib.contextmanager
+    def fetching_context(self):
+        """Hook to override to add context logic around batch fetching"""
+        yield
```

Comment:
how's this context expected to be used?

Reply:
This is used by the `InterBatchParallelismDataFetcher` to wrap the batch fetching logic within the cuda stream context.

```py
    @contextlib.contextmanager
    def fetching_context(self):
        """Wrap the batch fetching logic under a cuda stream"""
        with torch.cuda.stream(self.cuda_stream):
            yield
```